### DISCUSSION TITLE
How to use Causal Length Set?

### DISCUSSION THREAD

**Discussion by jeromegn** [vlcn-io/cr-sqlite#185] [Q&A] [answered]
The documentation refers to CLSets as a possible CRDT to use for tables, however I'm not sure how to actually enable that.

The examples show `CREATE CLSet table_name ...`, but that's not presently supported.

So my question is: how do I setup a table to use the causal length set CRDT?

**Comment by tantaman** [comment]
Currently only delete-wins is implemented and is the default when upgrading a table via `SELECT crsql_as_crr('table_name');`. I'll be adding causal length sets this quarter.

Along with that -- the syntax will get an update to no longer use a `SELECT` statement to modify the DB schema: https://github.com/vlcn-io/cr-sqlite/issues/181

I've updated the docs to correctly reflect the current state.

**Reply by tantaman** [reply]
deploying the docs now

commit: https://github.com/vlcn-io/docs/commit/1eb977bd6bef3b200111bc2c2fda61dac3eb0d2e

**Comment by jeromegn** [comment]
Thank you, that was a bit confusing. It seems like there's information about future features but it sounds like it already works.

I can live with delete-wins for now, but will it be changeable to CLSet in the future when it's available?

**Reply by tantaman** [reply]
I'll take another pass through the docs to make sure future features are clearly indicated.

> but will it be changeable to CLSet in the future when it's available?

I'd like to support that since people aren't going to model things exactly right the first time or may wish to evolve their schema.

I can't promise that it'll be supported as soon as CLSets are available and migrating between the two types might be another iteration afterwards.

**Comment by tantaman** [comment, marked as answer]
All tables are now modeled as "causal length sets" now. In other words, tables that have taken deletions are allowed have those deleted rows re-inserted at a later time by the user if they so desire.
